### PR TITLE
Fix for self-suppression in rest client: issue 42223

### PR DESCRIPTION
**Pull request by Fuud**
Fix for self-suppression in rest client: https://github.com/elastic/elasticsearch/issues/42223

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by swallez** [comment]
@elasticmachine ok to test

**Review by swallez — APPROVED** [review]
Looks good! Thanks for fixing this!

**Comment by swallez** [comment]
Fixes #42223

**Comment by swallez** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by swallez** [comment]
@elasticmachine update branch

**Comment by philkra** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by philkra** [comment]
Hi @Fuud , the PR is almost good to go, could you please add the following yaml file?

```
pr: 83568
summary: Fix for self-suppression in rest client
area: Clients
type: bug
issues:
 - 42223
```

This is used for curating the changelog, [ref](https://github.com/elastic/elasticsearch/pull/83955/commits/dcd2238e3775d5e5ffd682ef6b79eabb046812e5). 

Thank you very much for your contributions

**Comment by Fuud** [comment]
> Hi @Fuud , the PR is almost good to go, could you please add the `docs/changelog/83568.yaml` file with the following content?

Hi, done.

**Comment by technige** [comment]
@elasticmachine run elasticsearch-ci/part-2

File: docs/changelog/83568.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 83568
+summary: Fix for self-suppression in rest client
+area: Clients
```
Comment (philkra) [inline docs/changelog/83568.yaml:3]:
```suggestion
area: Client
```

[the remark this answers is not in the document]
Comment (philkra) [reply docs/changelog/83568.yaml]: @technige I updated to `Client`

**Comment by swallez** [comment]
Thanks for your contribution @Fuud! I'll take care of backporting this fix to other branches as the automated backport failed.

**Pull request merged**